akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -242,27 +242,47 @@ private[stream] object Collect {
   override def initialAttributes: Attributes = DefaultAttributes.collect and SourceLocation.forLambda(pf)
 
   def createLogic(inheritedAttributes: Attributes) =
-    new SupervisedGraphStageLogic(inheritedAttributes, shape) with InHandler with OutHandler {
-
-      import Collect.NotApplied
+    new GraphStageLogic(shape) with InHandler with OutHandler {
+      private lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
+      private var buffer: OptionVal[Out] = OptionVal.none
```

Comment:
Since it now became a detached/1 buffer stage I think we should also propagate context, just like Filter does (that's for telemetry to be able to follow the element through such stages)

Reply:
I dont know much about telemetry,and there are no scaladoc there too.

Reply:
try to add same kind of `contextPropagation` as in the Filter stage

Reply:
Would be better to have some scaladoc of `ContextPropagation`.